pkgs/development/libraries/libnixxml/default.nix

```diff
@@ -11,17 +11,36 @@ stdenv.mkDerivation {
     sha256 = "sha256-HKQnCkO1TDs1e0MDil0Roq4YRembqRHQvb7lK3GAftQ=";
   };
 
-  preConfigure = ''
+  postPatch = ''
     ./bootstrap
+
+    # Remove test that fails to prevent test failure.
```

Comment:
```suggestion
  preConfigure = ''
    ./bootstrap
  '';

  postPatch = ''
    # Remove test that fails to prevent test failure.
```

Reply:
That's the main change, but I'm not sure if running the `bootstrap` script in postPatch is really the right location. Running it in preConfigure is the cause of the build failure since (I assume) the autoreconfHook runs before the preConfigure hook. Not running bootstrap first causes the readme to not be generated, which caused a fail during the autoreconfHook.

Reply:
https://github.com/NixOS/nixpkgs/blob/7cb76200088f45cd24a9aa67fd2f9657943d78a4/pkgs/build-support/setup-hooks/autoreconf.sh#L1-L7

Maybe try this instead:

```nix
  preAutoreconf`= ''
    ./bootstrap
  '';
```

Reply:
Personally, I would use `preBuild` instead of `postPatch`. It is a build instruction that runs before the regular GNU Autotools build procedure starts.

Reply:
`preBuild` is even later than `preConfigure`. The problem is that the `autoreconfHook` errors out due to something `bootstrap` is supposed to do.

@Mindavi If you check the bootstrap script, you'll find the line that fixes your problem:

```sh
ln -s README.md README
```

Since bootstrap also invokes `autoreconf`, you should either

- do that symlink in `postPatch` / `preAutoreconf` and never call `bootstrap` or
- drop `autoreconfHook`, add regular `autoreconf` & whatever's needed and move `bootstrap` to `preConfigure`

Reply:
I chose to extract the symlinking and creation of the directory into the preAutoreconf hook, and not running the bootstrap script anymore. That seems most logical to me.